plugins/iamauth.js

```diff
@@ -153,8 +163,10 @@ class IAMPlugin extends BasePlugin {
                 callback(new Error('Invalid response from IAM token service'));
               }
             } else {
-              self.shouldApplyIAMAuth = false;
-              callback(new Error('Failed to access token'));
+              if (response.statusCode < 500 && response.statusCode !== 429) {
+                self.shouldApplyIAMAuth = false; // no retry
```

Comment:
I think we should output a message here (or where the error callback is consumed internally) since this never bubbles up to the user. Even if we add a flag to allow it to bubble up we should still log it more visibly than debug.

Reply:
Added a `debug('Disabling IAM authentication.');` in the callback.